File: lab1/zadanie3/zadanie3.py
def main():
	n = int(input("Podaj liczbę n: "))
	p = int(input("Podaj podstawę p (p>=2): "))
	a = n
	m = 0
	while a>0:
		m = p*m + a%p
		a = a//p
	if n==m:
		print(f"Liczba {n} o podstawie {p} jest palindromem")
	else:
		print(f"Liczba {n} o podstawie {p} nie jest palindromem")
	


if __name__ == "__main__":
	main()

File: lab1/zadanie3/zadanie3.c
#include <stdio.h>

int n, p, a, m;

int main(){
	printf("Podaj liczbę n: ");
	scanf("%d", &n);
	printf("Podaj podstawę p (p>=2): ");
	scanf("%d", &p);
	
	a = n;
	m = 0;
	
	while (a>0){
		m = p*m + a%p;
		a = a/p;
	}
	printf("Liczba %d o podstawie %d ", n, p);
	if (n==m){
		printf("jest palindromem");
	}
	else{
		printf("nie jest palindromem");
	}
}
